lib/data/external/AwsClient.js

```diff
@@ -15,6 +42,8 @@ class AwsClient {
         this._dataStoreName = config.dataStoreName;
         this._serverSideEncryption = config.serverSideEncryption;
         this._client = new AWS.S3(this._s3Params);
+        const awsGetPreCheck = _awsGetPreCheck.bind(this);
```

Comment:
why not define as a method on the class then?

Reply:
updated to create a `head` class method instead